sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,1045 @@
+ * SimpleFunction&lt;InputFormatKeyClass, MyKeyClass&gt; myOutputKeyType =
+ *       new SimpleFunction&lt;InputFormatKeyClass, MyKeyClass&gt;() {
+ *         public MyKeyClass apply(InputFormatKeyClass input) {
+ *           // ...logic to transform InputFormatKeyClass to MyKeyClass
+ *         }
+ *       };
+ * p.apply("read",
+ *     HadoopInputFormatIO.<MyKeyClass, InputFormatKeyClass>read()
+ *              .withConfiguration(myHadoopConfiguration)
+ *              .withKeyTranslation(myOutputKeyType);
+ * }
+ *
+ * // Read data with configuration and value translation (Example scenario: Beam Coder is not
+ * available for value class hence value translation is required.).
+ * SimpleFunction&lt;InputFormatValueClass, MyValueClass&gt; myOutputValueType =
+ *       new SimpleFunction&lt;InputFormatValueClass, MyValueClass&gt;() {
+ *         public MyValueClass apply(InputFormatValueClass input) {
+ *           // ...logic to transform InputFormatValueClass to MyValueClass
+ *         }
+ *       };
+ * p.apply("read",
+ *     HadoopInputFormatIO.<InputFormatKeyClass, MyValueClass>read()
+ *              .withConfiguration(myHadoopConfiguration)
+ *              .withValueTranslation(myOutputValueType);
+ * }</pre>
+ *
+ * <h3>Read data from Cassandra using HadoopInputFormatIO transform</h3>
+ *
+ * <pre>{@code
+ * Configuration cassandraConf = new Configuration();
+ * cassandraConf.set("cassandra.input.thrift.port", "9160");
+ * cassandraConf.set("cassandra.input.thrift.address", CassandraHostIp);
+ * cassandraConf.set("cassandra.input.partitioner.class", "Murmur3Partitioner");
+ * cassandraConf.set("cassandra.input.keyspace", "myKeySpace");
+ * cassandraConf.set("cassandra.input.columnfamily", "myColumnFamily");
+ * cassandraConf.setClass("key.class", {@link java.lang.Long Long.class}, Object.class);
+ * cassandraConf.setClass("value.class", {@link com.datastax.driver.core.Row Row.class}, Object.class);
+ * cassandraConf.setClass("mapreduce.job.inputformat.class", {@link org.apache.cassandra.hadoop.cql3.CqlInputFormat CqlInputFormat.class}, InputFormat.class);
+ * }</pre>
+ *
+ * <p>Call Read transform as follows:</p>
+ * <pre>{@code
+ * PCollection<KV<Long, String>> cassandraData =
+ *          p.apply("read",
+ *                  HadoopInputFormatIO.<Long, String>read()
+ *                      .withConfiguration(cassandraConf)
+ *                      .withValueTranslation(cassandraOutputValueType);
+ * }</pre>
+ *
+ * <p>The CqlInputFormat value class is {@link com.datastax.driver.core.Row Row}, which does not have a
+ * Beam Coder. Rather than write a new coder, you can provide your own translation method as
+ * follows:</p>
+ * <pre>{@code
+ * SimpleFunction<Row, String> cassandraOutputValueType = SimpleFunction<Row, String>(){
+ *    public String apply(Row row) {
+ *      return row.getString('myColName');
+ *    }
+ * };
+ * }</pre>
+ *
+ * <h3>Read data from Elasticsearch using HadoopInputFormatIO transform</h3>
```

Comment:
Concrete suggestion:
* Keep the only generic examples in the java doc
* move the cassandra/ES examples to a README.md in the root of this module. 

I think it might actually be handy to have examples for InputFormats that we know work - the README would let folks add it there easily and not feel like they were overwhelming the javadoc. We might move that content over to the website once we have per-IO pages (although I'd honestly prefer to point them at READMEs, but that's a whole other discussion.)